src/activities/morse_code/MorseCode.qml

```diff
@@ -86,8 +86,15 @@ ActivityBase {
             }
 
             function initSubLevel() {
+                if(dataset[currentLevel].values[0] == '_random_') {
+                    var randomIndex = Math.floor(Math.random() * dataset[currentLevel].values[1].length)
+                    questionValue = dataset[currentLevel].values[1][randomIndex]
+                    dataset[currentLevel].values[1].splice(randomIndex,1)
+                    score.numberOfSubLevels = dataset[currentLevel].values[2]
```

Comment:
Instead of storing the number of sublevels, I would directly use the list length (dataset[currentLevel].values[1].length if I'm not wrong). This way, if we add/remove letters, we won't have to not forget to update the size.